test/LibraryManager.IntegrationTest/FileSaveRestoreTests.cs

```diff
@@ -0,0 +1,344 @@
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"destination\":");
+            Editor.KeyboardCommands.Type("wwwroot/lib/jquery");
+
+            _libManConfig.Save();
+            string cwd = Path.Combine(Path.GetDirectoryName(_webProject.FullPath), "wwwroot", "lib", "jquery");
+            LibmanTestsUtility.WaitForRestoredFiles(cwd, expectedFilesAndFolders, caseInsensitive: true);
+
+            Editor.Caret.MoveToExpression("{", 0, 2);
+
+            for (int i = 0; i < 4; ++i)
+            {
+                Editor.Edit.DeleteLine();
+            }
+
+            _libManConfig.Save();
+            cwd = Path.Combine(Path.GetDirectoryName(_webProject.FullPath), "wwwroot", "lib");
+            LibmanTestsUtility.WaitForDeletedFile(cwd, "jquery", caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void FileSaveRestore_Cdnjs_DeleteFile()
+        {
+            _libManConfig.Open();
+            string[] expectedFilesAndFolders = new[] {
+                "jquery.js",
+                "jquery.min.js",
+            };
+
+            Editor.Caret.MoveToExpression("\"version\": \"1.0\"");
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "cdnjs", caseInsensitive: true);
+            Editor.KeyboardCommands.Type("cdnjs");
+
+            Editor.Caret.MoveToExpression("\"libraries\"");
+            Editor.Caret.MoveDown(2);
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true, timeout: 5000);
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"destination\":");
+            Editor.KeyboardCommands.Type("wwwroot/lib/jquery");
+
+            Editor.KeyboardCommands.Right();
+            Editor.KeyboardCommands.Type(",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"files\": [");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"jquery.js\",");
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"jquery.min.js\"");
+
+            _libManConfig.Save();
+            string cwd = Path.Combine(Path.GetDirectoryName(_webProject.FullPath), "wwwroot", "lib", "jquery");
+            LibmanTestsUtility.WaitForRestoredFiles(cwd, expectedFilesAndFolders, caseInsensitive: true);
+
+            Editor.Edit.DeleteToBeginningOfLine();
+            _libManConfig.Save();
```

Comment:
Won't this leave a comma after "jquery.js", which invalidates the json file?  I'm surprised this even works if the file doesn't parse.

Reply:
Looks like there's a bug make this comma work. However, I should not leave the extra comma here.

---
In reply to: [194557761](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194557761) [](ancestors = 194557761)